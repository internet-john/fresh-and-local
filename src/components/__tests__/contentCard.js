import React from 'react';
import renderer from 'react-test-renderer';

import ContentCard from '../content/contentCard';

describe('ContentCard', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ContentCard
          context={'HOME_PG'}
          image={``}
          data={{ header: 'Winter', content: 'December 21 - March 19 2020' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Displays header and content fields for season card', () => {
    const tree = renderer
      .create(
        <ContentCard
          context={'HOME_PG'}
          image={``}
          data={{ header: 'Winter', content: 'December 21 - March 19 2020' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Displays node name, type, general and nutritional info for produce card', () => {
    const mockProduceNode = {
      node: {
        name: 'Broccoli',
        type: 'Vegetable',
        generalInfo:
          'Eat it raw or cooked, and in dishes such as soups, stews, salads, or on its own.',
        nutritionInfo:
          'The nutrients in asparagus can support heart and bone health, while the folate and iron that it contains may be especially beneficial during pregnancy.',
      },
    };
    const tree = renderer
      .create(
        <ContentCard context={'SEASON_PG'} image={``} data={mockProduceNode} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
