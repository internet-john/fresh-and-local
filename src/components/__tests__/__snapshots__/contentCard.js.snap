// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard Displays header and content fields for season card 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-winter-2 MuiPaper-elevation1 MuiPaper-rounded"
>
  <a
    className="makeStyles-link-1"
    href="/winter/"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardMedia-root"
        style={
          Object {
            "backgroundImage": "url(\\"/static/images/cards/contemplative-reptile.jpg\\")",
          }
        }
        title="Contemplative Reptile"
      />
      <div
        className="MuiCardContent-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Winter
        </h5>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          December 21 - March 19 2020
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </a>
</div>
`;

exports[`ContentCard Displays node name, type for produce card 1`] = `null`;

exports[`ContentCard renders correctly 1`] = `null`;
